module/VuFind/src/VuFind/Search/Factory/AbstractSolrBackendFactory.php

```diff
@@ -182,7 +182,9 @@ public function __invoke(ContainerInterface $sm, $name, array $options = null)
      */
     protected function createBackend(Connector $connector)
     {
+        $config = $this->config->get($this->mainConfig);
         $backend = new $this->backendClass($connector);
+        $backend->setPageSize($config->Index->record_batch_size);
```

Comment:
If record_batch_size is not set in config.ini, this code will trigger a notice about an undefined value. I would suggest either wrapping the setPageSize() call in an `if (!empty(...)) {` check, or else providing a default value in the set call (i.e. `$config->Index->record_batch_size ?? 100`).